src/main/java/bio/terra/service/DatasetService.java

```diff
@@ -89,34 +83,25 @@ public EnumerateDatasetModel enumerate(
         return new EnumerateDatasetModel().items(summaries).total(datasetEnum.getTotal());
     }
 
-    public DeleteResponseModel delete(UUID id, AuthenticatedUserRequest userInfo) {
-        FlightMap flightMap = new FlightMap();
-        flightMap.put(JobMapKeys.REQUEST.getKeyName(), id);
-        flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(), "Deleting the dataset with ID " + id);
-        flightMap.put(JobMapKeys.USER_INFO.getKeyName(), userInfo);
-        String flightId = stairway.submit(DatasetDeleteFlight.class, flightMap);
-        return getResponse(flightId, DeleteResponseModel.class);
+    public DeleteResponseModel delete(UUID id, AuthenticatedUser userInfo) {
+        return jobService.submitAndWait(
+            "Delete dataset " + id,
+            DatasetDeleteFlight.class,
+            id,
+            userInfo,
+            DeleteResponseModel.class);
     }
 
-    public String ingestDataset(String id, IngestRequestModel ingestRequestModel) {
+    public String ingestDataset(String id, IngestRequestModel ingestRequestModel, AuthenticatedUser userInfo) {
         // Fill in a default load id if the caller did not provide one in the ingest request.
         if (StringUtils.isEmpty(ingestRequestModel.getLoadTag())) {
             String loadTag = "load-at-" + Instant.now().atZone(ZoneId.of("Z")).format(DateTimeFormatter.ISO_INSTANT);
             ingestRequestModel.setLoadTag(loadTag);
         }
-
-        FlightMap flightMap = new FlightMap();
-        flightMap.put(JobMapKeys.DESCRIPTION.getKeyName(),
+        String description =
             "Ingest from " + ingestRequestModel.getPath() +
                 " to " + ingestRequestModel.getTable() +
-                " in dataset id " + id);
-        flightMap.put(JobMapKeys.REQUEST.getKeyName(), ingestRequestModel);
-        flightMap.put(IngestMapKeys.DATASET_ID, id);
-        return stairway.submit(DatasetIngestFlight.class, flightMap);
-    }
-
-    private <T> T getResponse(String flightId, Class<T> resultClass) {
-        stairway.waitForFlight(flightId);
-        return jobService.retrieveJobResult(flightId, resultClass);
+                " in dataset id " + id;
+        return jobService.submit(description, DatasetIngestFlight.class, ingestRequestModel, userInfo);
```

Comment:
It looks like you dropped the dataset id out of the flightMap. How is that supposed to work?
It is not a valid assumption that the inputs are all in the incoming model.
Maybe we can come up with another way to DRY out setting up the common flight map stuff without mixing it with the submission step.